src/javascript/app/pages/user/account/settings/personal_details.js

```diff
@@ -54,6 +58,76 @@ const PersonalDetails = (() => {
         $('#missing_details_notice').setVisibility(!!has_missing_field);
     };
 
+    const populateChangeableFields = () => {
+        if (!has_changeable_fields) return;
+
+        const landing_companies = State.getResponse('landing_company');
+        const changeable        = landing_companies.financial_company.changeable_fields;
+        if (changeable && changeable.only_before_auth) {
+            changeable_fields = changeable.only_before_auth;
+        }
+    };
+
+    /**
+     * Remove labels and static fields and replace them with input when fields are changeable.
+     *
+     * @param {get_settings} to prepopulate some of the values.
+     */
+    const displayChangeableFields = (get_settings) => {
```

Comment:
can't we use `displayGetSettingsData` to also handle showing/hiding input for changeable fields? it feels like we're hardcoding a lot of fields in this function, while they are automated in `displayGetSettingsData`. it feels like repeating the work, but in a non-dynamic way.

Reply:
I can't think of a way that we can. Since I have to init data differently, we can't streamline it the way it has been done in `get_settings`. I would glad to refactor this if you have got some pointers to lead me to the right direction.